### DISCUSSION TITLE
Allow deleting dependent resources when they have not yet been created

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5890] [Feature Requests]
> Our users are using the clone and carry-forward options when creating a group of similar specimens into the system. They often need to delete a preparation or a few of the collection object property traits with each new specimen. However we haven't given these users any "delete" permissions as part of their roles.
> 
> Would it be possible for Specify to ignore the delete permissions when a specimen (or any new object) has yet to be saved and added to the database? It would make things a lot easier, and there is no danger of deleting anything that has been registered. Of course once they click save their role permission needs to activate again!

**Requested by:** CSIRO on [Asana](https://app.asana.com/0/1201972579963460/1205809442598595/f)

**Comment by maxpatiiuk** [comment]
hmm, I though we were already doing this, but maybe not as an oversight

we already have the logic for checking for "create" permission for new resources but "edit" for existing ones:

https://github.com/specify/specify7/blob/a0b59e34782bc66673944bc6e68c56f2e8b32067/specifyweb/frontend/js_src/lib/components/Forms/ResourceView.tsx#L81-L88

would look something similar, but for delete

Related, but removing independent resource doesn't require delete permission as rather than deleting the independent resource, it simply removes the reference to it:

https://github.com/specify/specify7/blob/a0b59e34782bc66673944bc6e68c56f2e8b32067/specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx#L171-L184

**Comment by grantfitzsimmons** [comment]
Mentioned as a big QOL improvement if we can resolve soon

**Comment by maxpatiiuk** [comment]
This might be a 5 line PR (excluding testing)

**Comment by zoewarner1** [comment]
Hi @grantfitzsimmons - just wondering if there was an update on this? It has come up again from a user and likely to have more users using the clone function soon too. Not being able to delete dependent resources that aren't needed is a bit of a sticking point. Thanks!

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk Can you tell us which 5 lines?

**Reply by maxpatiiuk** [reply]
I think this is the relevant line:

https://github.com/specify/specify7/blob/f292f99096dcc333d01751b54b8b729ba54dc7f3/specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx#L308-L311

For existing resources, check for "delete" permission like it does right now. But for not-yet-saved resources, the "read" should be sufficient (don't need to check for "create" as that is already assumed by the fact the user was able to add a new not-yet-saved resource)
